File: src/code-samples/langsmith/smithdb-migration/runs-query-basic-after.kt
///usr/bin/env jbang "$0" "$@" ; exit $?
//JAVA 21
//KOTLIN 2.0.21
//DEPS com.langchain.smith:langsmith-java:0.1.0-beta.8

// :snippet-start: smithdb-runs-query-basic-after-kt
// :codegroup-tab: After
import com.langchain.smith.client.LangsmithClient
import com.langchain.smith.client.okhttp.LangsmithOkHttpClient
import com.langchain.smith.models.runs.RunQueryV2Params

// :remove-start:
fun main() {
    if (System.getenv("LANGSMITH_API_KEY").isNullOrBlank()) {
        println("[smithdb-runs-query-basic-after] Skipping (LANGSMITH_API_KEY is not set).")
        return
    }

// :remove-end:
val client: LangsmithClient = LangsmithOkHttpClient.fromEnv()

val params = RunQueryV2Params.builder()
    .addProjectId("<project-id>")
    .runType(RunQueryV2Params.RunType.LLM)
    .filter("""and(eq(status, "success"), gt(total_tokens, 100))""")
    .addSelect(RunQueryV2Params.Select.ID)
    .addSelect(RunQueryV2Params.Select.NAME)
    .addSelect(RunQueryV2Params.Select.STATUS)
    .build()

client.runs().queryV2(params).autoPager().forEach { run ->
    println("${run.id()} ${run.name()} ${run.status()}")
}
// :remove-start:
}
// :remove-end:
// :snippet-end:

File: src/code-samples/langsmith/smithdb-migration/runs-query-basic-after.ts
// :remove-start:
if (!process.env.LANGSMITH_API_KEY) {
  console.log("Skipping (LANGSMITH_API_KEY required).");
  process.exit(0);
}
process.on("uncaughtException", (reason) => {
  const msg = String(reason);
  if (
    /\b[45]\d{2}\b/.test(msg) ||
    msg.includes("Invalid UUID") ||
    msg.includes("PermissionDenied")
  ) {
    console.log(`Skipping (placeholder values in test env): ${reason}`);
    process.exit(0);
  }
  process.exit(1);
});
// :remove-end:

// :snippet-start: runs-query-basic-after-js
// :codegroup-tab: After
import { Client } from "langsmith";

const client = new Client();
const project = await client.projects
  .list({ name: "default", limit: 1 })
  .then((page) => page.getPaginatedItems()[0]);
const page = client.runs.query({
  project_ids: [String(project.id)],
  run_type: "LLM",
  filter: 'and(eq(status, "success"), gt(total_tokens, 100))',
  selects: ["ID", "NAME", "STATUS"],
});
for await (const run of page) {
  console.log(run.id, run.name, run.status);
}
// :snippet-end:
